src/main/java/org/broadinstitute/hellbender/tools/spark/sv/CallVariantsFromAlignedContigsSAMSpark.java

```diff
@@ -0,0 +1,96 @@
+package org.broadinstitute.hellbender.tools.spark.sv;
+
+import com.google.common.annotations.VisibleForTesting;
+import htsjdk.samtools.util.SequenceUtil;
+import htsjdk.variant.variantcontext.VariantContext;
+import org.apache.commons.collections4.IterableUtils;
+import org.apache.commons.collections4.IteratorUtils;
+import org.apache.spark.api.java.JavaPairRDD;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.apache.spark.broadcast.Broadcast;
+import org.broadinstitute.hellbender.cmdline.Argument;
+import org.broadinstitute.hellbender.cmdline.CommandLineProgramProperties;
+import org.broadinstitute.hellbender.cmdline.programgroups.SparkProgramGroup;
+import org.broadinstitute.hellbender.engine.datasources.ReferenceMultiSource;
+import org.broadinstitute.hellbender.engine.filters.ReadFilter;
+import org.broadinstitute.hellbender.engine.filters.ReadFilterLibrary;
+import org.broadinstitute.hellbender.engine.spark.GATKSparkTool;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import scala.Tuple2;
+
+import java.util.List;
+import java.util.Optional;
+
+@CommandLineProgramProperties(summary="Parse SAM records and call SVs",
+        oneLineSummary="Parse SAM records and call SVs",
+        programGroup = SparkProgramGroup.class)
+public class CallVariantsFromAlignedContigsSAMSpark extends GATKSparkTool {
+    private static final long serialVersionUID = 1L;
+
+    @Argument(doc = "URL of the output path", shortName = "outputPath",
+            fullName = "outputPath", optional = false)
+    private String outputPath;
+
+    @Argument(doc = "FASTA formatted reference", shortName = "fastaReference",
+            fullName = "fastaReference", optional = false)
+    private String fastaReference;
+
+    @Argument(doc = "Minimum flanking alignment length", shortName = "minAlignLength",
+            fullName = "minAlignLength", optional = true)
+    private Integer minAlignLength = CallVariantsFromAlignedContigsSpark.DEFAULT_MIN_ALIGNMENT_LENGTH;
+
+    @Override
+    public boolean requiresReference() {
+        return true;
+    }
+
+    @Override
+    public boolean requiresReads() {
+        return true;
+    }
+
+    @Override
+    public ReadFilter makeReadFilter() {
+        return ReadFilterLibrary.MAPPED;
+    }
+
+    @Override
+    protected void runTool(final JavaSparkContext ctx) {
+        final Broadcast<ReferenceMultiSource> broadcastReference = ctx.broadcast(getReference());
+        final JavaPairRDD<Tuple2<String, String>, Iterable<GATKRead>> alignmentsGroupedByName = getReads().mapToPair(r -> new Tuple2<>(new Tuple2<>(r.getName(), r.getName()), r)).groupByKey();
+        final JavaPairRDD<Tuple2<String, String>, Tuple2<byte[], Iterable<AlignmentRegion>>> alignmentRegionsIterable = alignmentsGroupedByName.mapValues(CallVariantsFromAlignedContigsSAMSpark::convertToAlignmentRegions);
```

Comment:
If you're talking about the second entry in the `alignmentsGroupedByName` tuple, it's because this includes supplementary alignments so you'll get multiple `GATKRead` objects for each read name.